pkg/apis/core/v1alpha1/workflow_types.go

```diff
@@ -0,0 +1,83 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// WorkflowSpec defines the desired state of Workflow.
+type WorkflowSpec struct {
+	// Name of the Template associated with this workflow.
+	TemplateRef string `json:"templateRef,omitempty"`
+
+	// A mapping of template devices to hadware mac addresses
+	HardwareMap map[string]string `json:"hardwareMap,omitempty"`
+}
+
+// WorkflowStatus defines the observed state of Workflow.
+type WorkflowStatus struct {
+	// State is the state of the workflow in Tinkerbell.
+	State string `json:"state,omitempty"`
+
+	// Data is the populated Workflow Data in Tinkerbell.
+	Data string `json:"data,omitempty"`
```

Comment:
Should we drop this? Its just the evaluated, interpolated template with the hardware mappings. This is all encoded and strongly typed in the `Tasks Task[]` field below